### DISCUSSION TITLE
LSM6DSL - How to calculate G Force

### DISCUSSION THREAD

**Discussion by cherbin** [stm32duino/Arduino_Core_STM32#2529] [Q&A] [answered]
Hello,
Using the https://github.com/stm32duino/LSM6DSL examples

How is the G-Force calculated ? 
Specifically  Lateral and Longitudinal G-Force ?

**Comment by fpistm** [comment]
Hi 
@cparata could you answer, please?

**Comment by cparata** [comment, marked as answer]
Hi @cherbin ,
with the LSM6DSL library you can calculate the acceleration and the angular rate. The acceleration and the angular rate are both vector quantities, so you have the measurements on three coordinates (x, y, z). The acceleration is expressed in "milli-g" where g is the gravity acceleration (9,81 m/s^2). So, typically, if the board of the accelerometer is still on a table with the z axis perpendicular to the table, you will get a measurement using the "Get_X_Axes" API like (~+/-0, ~+/-0, ~+/-1000) because the only force that acts on the device is the gravity force and so the acceleration on the 3 axes is ~(0, 0, 9,81m/s^2) = (0, 0, 1g) = (0, 0, 1000 milli-g). Actually, this is the ideal measurement; with the real device you have some approximations and electrical noise that give you more or less the same result with some errors well documented in the Datasheet of the device. Regarding the gyroscope, the measurement is expressed in milli-degree-per-second (it's an angular rate). Typically, if the board is still, you should get a measurement using the "Get_G_Axes" API like (~0, ~0, ~0) but with a real device you will get more or less this value with some errors that are well documented in the Datasheet of the device as well. If you start rotating the board, obviously the gyroscope measurements will change (and also the accelerometer measurements).
I hope that it can clarify your doubts.
Best Regards,
Carlo
